File: lib/src/structuredfield/decodeSfList.ts
import type { SfDecodeOptions } from './SfDecodeOptions';
import type { SfMember } from './SfMember';
import { parseError } from './parse/parseError.ts';
import { parseList } from './parse/parseList.ts';
import { LIST } from './utils/LIST.ts';

/**
 * Decode a structured field string into a structured field list
 *
 * @param input - The structured field string to decode
 * @returns The structured field list
 *
 * @group Structured Field
 *
 * @beta
 */
export function decodeSfList(input: string, options?: SfDecodeOptions): SfMember[] {
	try {
		const { src, value } = parseList(input.trim(), options);
		if (src !== '') {
			throw parseError(src, LIST);
		}
		return value;
	}
	catch (cause) {
		throw parseError(input, LIST, cause);
	}
}

File: python/src/structuredfield/decodeSfList.py
from .parse.parseError import parseError
from .parse.parseList import parseList
from .SfException import SfException
from .utils.LIST import LIST


def decodeSfList(input):
    """
    Decode a structured field string into a structured field list

    @param input - The structured field string to decode
    @returns The structured field list

    @group Structured Field

    @beta
    """
    try:
        list = parseList(input.strip())
        if list.src != "":
            raise parseError(list.src, LIST)

        return list.value

    except SfException as cause:
        raise parseError(input, LIST, cause)
